src/iris/api.py

```diff
@@ -1319,9 +1325,10 @@ class Messages(object):
 
     def on_get(self, req, resp):
         fields = req.get_param_as_list('fields')
-        if fields is None:
-            fields = message_columns
         req.params.pop('fields', None)
+        fields = [f for f in fields if f in message_columns]
```

Comment:
Why not do something similar to this instead?

`fields = set(fields) & set(message_columns)`

That will get only the columns present in both and is likely more efficient than the list comprehension that iterates through fields while checking for presence in message_columns.